### PR TITLE
Tradier live updates

### PR DESCRIPTION
**Pull request by davidlatte**
This pull request includes several changes to the `lumibot` project, focusing on improving order handling, parsing, and testing. The most important changes involve modifying the order parsing logic, updating the order retrieval methods, and enhancing the test coverage for the `Tradier` broker.

### Improvements to order handling and parsing:

* [`lumibot/brokers/broker.py`](diffhunk://#diff-d2068bcf1d622e9a1e7c91bfdf3ed70b4fdfaf47ab427d6b4791f45b0a3c21a3L260-R268): Updated the `_pull_broker_all_orders` method to return a list of dictionaries instead of `Order` objects.
* [`lumibot/brokers/tradier.py`](diffhunk://#diff-e3bd4675965d31b178a2669811a86df60b079840c334acfce1ee21d3b70c5ad1L338-R339): Modified the `_submit_order` method to correctly handle `STOP_LIMIT` orders and round the limit and stop prices to two decimal places. [[1]](diffhunk://#diff-e3bd4675965d31b178a2669811a86df60b079840c334acfce1ee21d3b70c5ad1L338-R339) [[2]](diffhunk://#diff-e3bd4675965d31b178a2669811a86df60b079840c334acfce1ee21d3b70c5ad1L353-R355)
* [`lumibot/brokers/tradier.py`](diffhunk://#diff-e3bd4675965d31b178a2669811a86df60b079840c334acfce1ee21d3b70c5ad1L590-R602): Enhanced the `_parse_broker_order` method to handle OCO (One Cancels the Other) orders by extracting values from child orders when necessary. Introduced the `_extract_order_value` helper method to facilitate this process. [[1]](diffhunk://#diff-e3bd4675965d31b178a2669811a86df60b079840c334acfce1ee21d3b70c5ad1L590-R602) [[2]](diffhunk://#diff-e3bd4675965d31b178a2669811a86df60b079840c334acfce1ee21d3b70c5ad1R637-R653)

### Enhancements to order retrieval methods:

* [`lumibot/brokers/tradier.py`](diffhunk://#diff-e3bd4675965d31b178a2669811a86df60b079840c334acfce1ee21d3b70c5ad1L653-R692): Added the `_clean_order_records` method to clean and round order records before converting them to dictionaries. This method is now used in `_pull_broker_all_orders`. This ensures consistant None type values for fields that don't apply to certain orders (i.e. stop_price for non-stop orders)
* [`lumibot/brokers/tradier.py`](diffhunk://#diff-e3bd4675965d31b178a2669811a86df60b079840c334acfce1ee21d3b70c5ad1R637-R653): Updated the `_pull_broker_order` method to use the `_clean_order_records` method for cleaning order data.

### Test coverage improvements:

* [`tests/test_tradier.py`](diffhunk://#diff-090fb263614b0554db6c9363865874867f3cefc68c8f0bc5350bcf2a5722f579R234-R286): Added a new test `test_pull_broker_all_orders` to verify the functionality of the `_pull_broker_all_orders` method.
* [`tests/test_tradier.py`](diffhunk://#diff-090fb263614b0554db6c9363865874867f3cefc68c8f0bc5350bcf2a5722f579R345-R439): Introduced a new test `test_oco_parse_broker_order` to ensure correct parsing of OCO orders.
* [`tests/test_tradier.py`](diffhunk://#diff-090fb263614b0554db6c9363865874867f3cefc68c8f0bc5350bcf2a5722f579R299): Updated existing tests to include the new `class` field in the order response to reflect the changes in the order parsing logic. [[1]](diffhunk://#diff-090fb263614b0554db6c9363865874867f3cefc68c8f0bc5350bcf2a5722f579R299) [[2]](diffhunk://#diff-090fb263614b0554db6c9363865874867f3cefc68c8f0bc5350bcf2a5722f579R479)

These changes collectively enhance the robustness of the order handling and parsing mechanisms in the `lumibot` project while ensuring comprehensive test coverage.

### What change is being made?
Enhance Tradier broker integration by refining order parsing logic, adding new utility functions for data cleaning, modifying order status handling, and updating corresponding unit tests.

### Why are these changes being made?
This update improves the accuracy of order handling by ensuring values are correctly rounded and missing values are managed appropriately, particularly for One-Cancels-Other (OCO) orders. The changes also extend the test suite to cover more cases, ensuring robust functionality. The inclusion of these enhancements aims to handle edge cases and improve reliability when processing broker orders, especially when the system restarts or trades complex order types like OCOs.

> Is this description stale? Ask me to generate a new description by commenting `/korbit-generate-pr-description`

### CODE REVIEW

File: lumibot/brokers/tradier.py [hunk 1]
```diff
@@ -650,7 +669,27 @@ def _pull_broker_all_orders(self):
...
+        ----------
+        df : pandas.DataFrame
+            The DataFrame containing order records.
+
+        Returns
+        -------
+        list[dict]
+            A list of dictionaries representing the cleaned order records.
+        """
+        rounded_df = df.applymap(lambda x: round(x, 2) if isinstance(x, float) else x)
```
[the remark this answers is not in the document]
Comment (davidlatte) [reply lumibot/brokers/tradier.py:690]: I like this!

**Review by grzesir — APPROVED** [review]
